File: javascript/6324.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");

input.slice(1).forEach((line, i) => {
  const [protocol, url] = [line.split("://")[0], line.slice(line.indexOf("://") + "://".length)];
  const [host, port] = url.split("/")[0].split(":");
  const path = url.indexOf("/") > 0 ? url.slice(url.indexOf("/") + 1) : "";

  console.log(`URL #${i + 1}`);
  console.log(`Protocol = ${protocol}`);
  console.log(`Host     = ${host}`);
  console.log(`Port     = ${port || "<default>"}`);
  console.log(`Path     = ${path || "<default>"}`);
  console.log("");
});

File: python/6324.py
from urllib.parse import urlparse

for i in range(int(input())):
    # fmt: off
    protocol, hostname, path, _, query, fragment = urlparse(url := input())
    # fmt: on
    host = (hostname := hostname.split(":"))[0]
    port = hostname[1] if len(hostname) > 1 else ""
    url = url[url.index("://") + len("://") :]
    path = (url[url.index("/") + 1 :] if url.find("/") else "") if query or fragment else path[1:]

    print(f"URL #{i+1}")
    print(f"Protocol = {protocol}")
    print(f"Host     = {host}")
    print(f"Port     = {port if port else '<default>'}")
    print(f"Path     = {path if path else '<default>'}")
    print("")
